iceoryx_posh/test/moduletests/test_posh_runtime_single_process.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2021 by Robert Bosch GmbH, Apex.AI Inc. All rights reserved.
```

Comment:
Here you should use TYPED_TESTS. The reason is that `PoshRuntimeSingleProcess` inherits from `PoshRuntime` therefore every `PoshRuntime` test case has to also run successfully for `PoshRuntimeSingleProcess`

The idea would be to replace the `PoshRuntime` test cases with TYPED_TESTS and test in this file only the cases which are unique to `PoshRuntimeSingleProcess`. To see them in action you could look at iceoryx_utils/test/moduletests/test_lockfree_queue.cpp

Reply:
Seems like you are proposing a refactoring of  posh runtime test file. It can be done in another PR :)

Reply:
@JimmyBch this is fine by me.

Reply:
I have difficulty to use TYPED_TESTS as PoshRuntime and PoshRuntimeSingleProcess do not have the same instantiation.

PoshRuntime Instantiation
const iox::ProcessName_t m_runtimeName{"publisher"};
PoshRuntime* m_runtime{&PoshRuntime::initRuntime(m_runtimeName)};

 PoshRuntimeSingleProcess
PoshRuntimeSingleProcess* m_runtime("publisher") ;

Do you have any hint to create a template config as in iceoryx_utils/test/moduletests/test_lockfree_queue.cpp in order to set correctly m_runtime depending of the poshRuntime class ?